web3c/confidential.js

```diff
@@ -10,89 +12,107 @@ const Signer = require('./signer');
-    };
+  setupConfidentialRpc(web3) {
+    let methods = [
+      // Second parameter - the long-term key - is intercepted by the provider.
+      new web3.extend.Method({
+        name: 'getPublicKey',
+        call: 'confidential_getPublicKey',
+        params: 1,
+        inputFormatter: [web3.extend.formatters.inputAddressFormatter],
+        outputFormatter: getPublicKeyOutputFormatter
+      }),
+      new web3.extend.Method({
+        name: 'call',
+        call: 'confidential_call_enc',
+        params: 2,
+        inputFormatter: [
+          web3.extend.formatters.inputCallFormatter,
+          web3.extend.formatters.inputDefaultBlockNumberFormatter
+        ],
+        outputFormatter: callOutputFormatter
+      }),
+    ];
+
+    methods.forEach((method) => {
+      method.setRequestManager(web3._requestManager);
+      method.attachToObject(this);
+    });
+  }
+
+  /**
+   * Creates the confidential `Contract` constructor on the `confidential` namespace.
+   * The `Contract` behaves in the same way as the eth `Contract` object, except
+   * transparently encrypts/decrypts all outbound/inbound requests according
+   * to the web3c spec.
+   */
+  setupConfidentialContract(web3, storage, mraebox) {
+    // Save `this` so that we can refer to it and its properties inside `ConfidentialContract`.
+    // Otherwise `this` is overridden when `new` is used in `new Contract`.
+    let self = this;
+
+    this.Contract = ContractFactory.make(web3, (options) => {
+      let provider = new ConfidentialProvider(this.keyManager, web3._requestManager);
+
+      let keymanager = self.keyManager;
+
+      if (options && options.saveSession === false) {
+        keymanager = new KeyManager(web3, undefined, mraebox);
+        provider = new ConfidentialProvider(keymanager, web3._requestManager);
+      }
 
-    if (options && options.key) {
-      keymanager.add(address, options.key);
-    }
-  };
+      if (options && options.key) {
+        keymanager.add(address, options.key);
+      }
 
-  this.resetKeyManager = () => {
+      return provider;
+    });
+
+  }
+
+  resetKeymanager() {
     this.keyManager.reset();
-  };
-};
-
-function wrapAccounts (accounts, transformer) {
-  let wrappedSigner = accounts.signTransaction.bind(accounts);
-
-  accounts.signTransaction = function signConfidentialTransaction (tx, from) {
-    if (tx.to) {
-      return new Promise(function (resolve, reject) {
-        transformer.encryptTx(tx, function finishSignConfidentialTransaction(err) {
-          if (err) {
-            reject(err);
-          }
-          try {
-            return wrappedSigner(tx, from).then(resolve, reject);
-          } catch (e) {
-            reject(e);
-          }
-        }); 
-      });
-    }
-    // deployment
-    tx.data = transformer._prependConfidential(tx.data);
-    return wrappedSigner(tx, from);
-  };
+  }
+
+  setupWrappedAccounts(web3) {
```

Comment:
these setup methods are public methods on the confidential object when declared this way. looks like JS doesn't currently have a way to make private class methods. these should be private though...

Reply:
They should be private, but this structure reads much more nicely imo. Is there a better solution than just adding  _?